### PR TITLE
Upgrade to the latest solidus_dev_support

### PR DESCRIPTION
**Pull request by MinasMazar**
Ref.  https://github.com/solidusio-contrib/solidus_dev_support/issues/10

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Copyright (c) 2020 [name of plugin creator]
```
Comment (kennyadsl) [inline LICENSE]: Let's put something here!

File: app/assets/javascripts/spree/backend/solidus_multi_domain.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Placeholder manifest file.
```
Comment (kennyadsl) [inline app/assets/javascripts/spree/backend/solidus_multi_domain.js]: We can remove empty files

File: lib/generators/solidus_multi_domain/install/install_generator.rb [hunk 3]
```diff
@@ -6,19 +6,19 @@ class InstallGenerator < Rails::Generators::Base
       class_option :auto_run_migrations, type: :boolean, default: false
 
       def add_javascripts
-        append_file "vendor/assets/javascripts/spree/backend/all.js", "//= require spree/backend/solidus_multi_domain\n"
+        append_file 'vendor/assets/javascripts/spree/backend/all.js', "//= require spree/backend/solidus_multi_domain\n"
```
Comment (kennyadsl) [inline lib/generators/solidus_multi_domain/install/install_generator.rb]: no need to install empty files on host applications, let's remove this method.

File: lib/solidus_multi_domain/engine.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      ApplicationController.include SolidusMultiDomain::MultiDomainHelpers
+    end
+
+    config.to_prepare &method(:activate).to_proc
+
+    initializer 'spree.promo.register.promotions.rules' do |app|
+      app.config.spree.promotions.rules << ::Spree::Promotion::Rules::Store
+    end
+
+    def self.admin_available?
```
Comment (kennyadsl) [inline lib/solidus_multi_domain/engine.rb]: Who is using these methods?

File: app/assets/javascripts/spree/backend/solidus_multi_domain.js (hunk 2 above)
Comment (elia) [reply app/assets/javascripts/spree/backend/solidus_multi_domain.js]: It has a require_tree at the bottom that will add `store_picker.js` as a dependency

File: lib/generators/solidus_multi_domain/install/install_generator.rb (hunk 3 above)
Comment (elia) [reply lib/generators/solidus_multi_domain/install/install_generator.rb]: The file added as a dependency has a `require_tree` that is not very visible and you might have missed 🔍

File: lib/solidus_multi_domain/engine.rb (hunk 4 above)
Comment (elia) [reply lib/solidus_multi_domain/engine.rb]: The extension only decorates some classes if the corresponding framework is available

image

Comment (kennyadsl) [reply lib/solidus_multi_domain/engine.rb]: Got it. We already have the same methods in [`solidus_support`](https://github.com/solidusio/solidus_support/blob/master/lib/solidus_support.rb#L52), which is a dependency already. 

Anyway, I don't think it is the right approach. 

We created this [`solidus_support`](https://github.com/solidusio/solidus_support/blob/cd4892d959d49f557bf13832c68f64dfcb113746/lib/solidus_support/engine_extensions.rb) module. Including that module will be just a matter of adding these decorators in the "right" folder, as done [here](https://github.com/solidusio/solidus_auth_devise/tree/master/lib/decorators). 

The module is not yet been released (we are waiting for some test in real extensions) so you can use it only by using the github version for now. You can make a test to see if everything works correctly or we can just switch to the `solidus_support` `*_available?` methods and face this change in a future PR, up to you!

File: app/assets/javascripts/spree/backend/solidus_multi_domain.js (hunk 2 above)
Comment (kennyadsl) [reply app/assets/javascripts/spree/backend/solidus_multi_domain.js]: Can't we remove this `require_tree` statement?

File: lib/solidus_multi_domain/engine.rb (hunk 4 above)
Comment (kennyadsl) [reply lib/solidus_multi_domain/engine.rb]: This is the PR for `solidus_auth_devise` to start using this approach: https://github.com/solidusio/solidus_auth_devise/pull/182

Comment (elia) [reply lib/solidus_multi_domain/engine.rb]: I would prefer not to mix the two changes, after all this PR is targeted at just upgrading the files to the latest prescriptions of solidus_dev_support and make the specs pass. 
I opened a separate issue as you suggested to keep track of this, and possibly use this extension as the testbed to validate that solidus_support PR

https://github.com/solidusio-contrib/solidus_multi_domain/issues/139

**Pull request merged**